torch_geometric/utils/isolated.py

```diff
@@ -7,19 +7,72 @@
 from torch_geometric.utils.num_nodes import maybe_num_nodes
 
 
+def num_isolated_nodes(
+    edge_index: Tensor,
+    num_nodes: Optional[Union[int, Tuple[int, int]]] = None,
+) -> Union[int, Tuple[int, int]]:
+    r"""Returns the number of isolated nodes in the graph given by
+    :attr:`edge_index`. Please specify :obj:`num_nodes` as a tuple of two
+    integers if the graph is bipartite. For normal graph, self loops are
+    automatically removed before checking for isolated nodes.
+
+    Args:
+        edge_index (LongTensor): The edge indices.
+        num_nodes (int or tuple, optional): The number of nodes, *i.e.*
+            :obj:`max_val + 1` of :attr:`edge_index`. For bipartite graph,
+            provide a tuple of two integers `(num_src_nodes, num_dst_nodes)`.
+            (default: :obj:`None`)
+
+    :rtype: int or tuple
+
+    Examples:
+        >>> edge_index = torch.tensor([[0, 1, 0],
+        ...                            [1, 0, 0]])
+        >>> num_isolated_nodes(edge_index)
+        0
+        >>> bi_edge_index = torch.tensor([[0, 1],
+        ...                               [1, 0]])
+        >>> num_isolated_nodes(bi_edge_index, num_nodes=(2, 2))
+        (0, 0)
+        >>> num_isolated_nodes(bi_edge_index, num_nodes=(2, 3))
+        (0, 1)
+        >>> num_isolated_nodes(bi_edge_index, num_nodes=(3, 3))
+        (1, 1)
+    """
+    if not isinstance(num_nodes, tuple):
+        num_nodes = maybe_num_nodes(edge_index, num_nodes)
+        edge_index, _ = remove_self_loops(edge_index)
+        return num_nodes - torch.unique(edge_index.view(-1)).numel()
```

Comment:
```suggestion
        return num_nodes - edge_index.view(-1).unique().numel()
```

Reply:
We need to use torch.unique instead of Tensor.unique to support torch jit. Otherwise we will have following error
```
E           RuntimeError:
E           'Tensor' object has no attribute or method 'unique'.:
E             File "/Users/zecheng/code/pytorch_geometric/torch_geometric/utils/isolated.py", line 57
E               else:
E                   num_src_nodes, num_dst_nodes = num_nodes
E                   num_src_isolated = num_src_nodes - edge_index[0].unique().numel()
E                                                      ~~~~~~~~~~~~~~~~~~~~ <--- HERE
E                   num_dst_isolated = num_dst_nodes - edge_index[1].unique().numel()

```